### ISSUE TITLE
Anti-aliasing as an FX node

### ISSUE THREAD

**Issue by WolfInABowl** [opentoonz/opentoonz#3247]
This is a request for the addition of an anti-aliasing FX node.

OpenToonz is already capable of applying anti-aliasing to raster images.
I am assuming (perhaps wrongfully so) that it would not be hugely difficult to create an FX node that utilizes at least some of this same code.

OpenToonz applies anti-aliasing at a fixed point in the rendering process, this means that the user is unable to work with the aliased source image.
Plugins to smooth images have been created for After Effects, these plugins are able to be placed anywhere in a layer's effects stack. This allows the user to work with the aliased source images for as long as they need to.

Here is an example of something that would be far harder or outright impossible to achieve in OT compared to After Effects:
![AE Blending](https://user-images.githubusercontent.com/42905003/80047949-a99a9680-8506-11ea-824d-28faa41f05f1.gif)

You can see that the white square is being blended with Dwanko's clothing, these are two separate images that have been layered on top of each other.
As the blending effect works based on pixel colors and is designed to work with aliased images, such as result is possible.
Finally, anti-aliasing is applied, producing a smoothed result.
Without the ability to apply effects to the aliased source image, getting artifact-free results would be difficult due to the increased amount of colors produced by anti-aliasing.

In conclusion, having the ability to apply anti-aliasing when needed, enables a workflow that is designed around the ability to work with aliased images.
Such a workflow can be seen in the Japanese Animation Industry where After Effects is commonly used for compositing.
Having more direct access to the pixel data of images allows for more complex compositing and VFX tasks to be achieved.
There are tradeoffs between OT's approach and the approach used in the above example:
OT, though more limited, does a good job of dealing with anti-aliased images as long as they are ToonzRaster or ToonzVector.
After Effects, though lacking the ability to work as cleanly with anti-aliased images, is more flexible as it is designed around manipulating regular raster images.

**Comment by gab3d** [comment]
It'd certainly increase the processing possibilities of the FX Schematic. 👍

**Comment by RodneyBaker** [comment]
While not quite implemented in the way outlined here there is now an antialiasing FX implemented via #6150 

Closing, although with anticipation this request will be updated/replaced depending on any shortfalls identified.
